src/components/buttons/Button/Button.jsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Icon } from "../../icons/Icon/Icon";
+
+import "./button.scss";
+
+/**
+ * Button
+ *
+ * All buttons used in the app
+ *
+ * @return {jsx}
+ */
+export const Button = ({
+  /* Add props here */
+  text,
+  iconName,
+  size,
+  color,
+  secondary,
+  ghost,
+  emergency,
+  disabled,
+  onClick,
+  ...rest
+}) => {
+  const mode = secondary
+    ? "btn--secondary"
+    : ghost
+    ? "btn--ghost"
+    : emergency
+    ? "btn--emergency"
+    : "btn--primary";
+
+  return (
+    <button
+      className={[
+        `btn btn--${size} btn--${color}`,
+        mode,
+        disabled && "btn--disabled",
+      ].join(" ")}
+      onClick={onClick}
+      {...rest}
+    >
+      {emergency ? (
+        <Icon name="phone-emergency" color={"#ffffff"} size={"large"} />
+      ) : iconName ? (
+        <Icon name={iconName} color={"#ffffff"} size={"medium"} />
+      ) : null}
+      {text ? <span>{text}</span> : null}
+    </button>
+  );
+};
+
+Button.propTypes = {
+  /**
+   * Text to render in the Button component
+   * */
+  text: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+
+  /**
+   * Icon to render in the Button component
+   **/
+  icon: PropTypes.oneOfType([PropTypes.element, PropTypes.bool]),
```

Comment:
This does not match the "iconName" prop.

Reply:
@georgiganchev-7DIGIT Please change this to "iconName" and also its type should be String as that's what we will be passing to it as far as I understand it.